File: 2530-maximal-score-after-applying-k-operations/2530-maximal-score-after-applying-k-operations.cpp
class Solution {
public:
    long long maxKelements(vector<int>& nums, int k) {
        priority_queue<long long> pq;
        for(int num : nums) pq.push(num);
        long long score = 0;
        while(k != 0){
            long long num = pq.top();
            pq.pop();
            score += num;
            num = ceil(num/3.0);
            pq.push(num);
            k--;
        }

        return score;
    }
};

File: 2530-maximal-score-after-applying-k-operations/2530-maximal-score-after-applying-k-operations.java
class Solution {
    public long maxKelements(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for(int num : nums) pq.add(num);

        long score = 0;

        while(k-- > 0){
            int num = pq.poll();
            score += num;
            num = (int)Math.ceil(num/3.0);
            pq.add(num);
        }

        return score;
    }
}